### PR TITLE
Forward port changes from 0.12.1 branch

### PR DESCRIPTION
**Pull request by Shimuuar**
What is ported:

 - Changelog updates
 - Transition from test-framework to tasty
 - mkType (#287) is restored but deprecated
 - New tests for maxIndex & Co. I had to change them because of change of behavior

### CODE REVIEW

**Comment by cartazio** [comment]
thanks! i'll take a look at this soon 
(this spring has been nutty! )

**Comment by cartazio** [comment]
i'll read more closely, but it looks like aside from some tweaking to retain the semantics checking flavor for the maxIndex etc stuff that i can easily add on top, should be good (though i've not read super duper closely yet)

File: changelog.md [hunk 1]
```diff
@@ -9,39 +9,7 @@ Changes in version next
...
-     not guaranteed that the `Storable` instances between two
-     representationally equal types will preserve this internal structure.
-     One consequence of this choice is that it is no longer possible to
-     `coerce` between `Storable.Vector a` and `Storable.Vector b` if `a` and
-     `b` are nominally distinct but representationally equal types. We now
-     provide `unsafeCoerce{M}Vector` functions in
-     `Data.Vector.Storable{.Mutable}` to allow this (the onus is on the user
-     to ensure that no `Storable` invariants are broken when using these
-     functions).
+
```
Comment (cartazio) [inline changelog.md]: in isolation, this looks wrong, but again, ill do the proper close reading tomorrow to check

Comment (Shimuuar) [reply changelog.md]: It was added in #235 I just copy pasted it without any changes

**Comment by lehins** [comment]
I said on multiple occasions that this is the kind of PRs can be completely avoided if original work was submitted as a PR with master as a target instead of a release branch. If @Shimuuar didn't spot it, next release could have easily be created without these changes.

**Comment by cartazio** [comment]
i suspect the cleanup can be made a tad simpler, i'll poke at it later today :) 

you're absolutely right about hind sight!

**Comment by Shimuuar** [comment]
@lehins approach you proposed is very good one but wouldn't quite work in this case since few commits were reverted in v0.12.1 branch

**Comment by cartazio** [comment]
yeah,  either way some surgery is needed.

**Comment by Shimuuar** [comment]
@cartazio Should this PR be merged?

**Comment by cartazio** [comment]
Derp I’ll look today or tomorrow.  Got derailed by other stuff

**Comment by cartazio** [comment]
@Shimuuar  I have a hopefully simpler version of this fwd port i'm going to push to a branch for PR later today or tomorrow (a bit frazzled today by all the stuff on fire/firearms stuff within 200 feet of where i live in the past few days ), was refining a test for the min/maxBy property test stuff and found a weakness in the default quickcheck instance for float and double that makes it "fail" because the QC generators for those two dont have the right properties for checking actual code 

https://github.com/nick8325/quickcheck/issues/295 <-- for context , the needed distribution to get good QC property tests on float /double is kinda fascinating!

@lehins @Shimuuar i'm genuinely sorry i've been kinda a neurotic control maven, i'm working on gettin better, and I hope we can all work together better more :) 

@lehins i'm sorry i wasnt following up on your emails, depending on my head space, a long email makes me anxious and then i avoid reading it. (old habbits for trying to avoid getting sucked into internet arguments)

i genuinely mean well, and the best way to asychronously get a fun positive 1-1 where I explain or try to unpack what i sometimes fail to communicate clearly is just to msg me asynchronously on freenodeIRC and i'll read it later and we can have a slow mo back an dforth there. (seriously, i'm sorry for being a frustrating collaborator at times,)

**Comment by cartazio** [comment]
deerp, i fliped a property around, i'll fix it now

**Comment by Shimuuar** [comment]
Superceded by #309

**Pull request closed without merging**